cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -1165,6 +1165,219 @@ int32_t gdv_fn_cast_intervalyear_utf8_int32(int64_t context_ptr, int64_t holder_
   auto* holder = reinterpret_cast<gandiva::IntervalYearsHolder*>(holder_ptr);
   return (*holder)(context, data, data_len, in1_validity, out_valid);
 }
+
+GANDIVA_EXPORT
+const char* translate_utf8_utf8_utf8(int64_t context, const char* in, int32_t in_len,
```

Comment:
How is replace different from translate?

Reply:
This implementation follow the PostGreSQL Translate example (used in Hive), this is not equals replace function, have some different rules

Look this to see all rules and examples: [PostgreSQL Translate](https://www.w3resource.com/PostgreSQL/translate-function.php)

If use replace for some cases, the result returned will be wrong
especially if there is repetition in **from_string** and different matches in **to_string**